src/activities/binary_bulb/BinaryBulb.qml

```diff
@@ -0,0 +1,169 @@
+ *   along with this program; if not, see <http://www.gnu.org/licenses/>.
+ */
+import QtQuick 2.6
+import GCompris 1.0
+
+import "../../core"
+import "binary_bulb.js" as Activity
+
+ActivityBase {
+    id: activity
+
+    onStart: focus = true
+    onStop: {}
+
+    pageComponent: Image {
+        id: background
+        anchors.fill: parent
+        source: "resource/background.svg"
+        signal start
+        signal stop
+
+        Component.onCompleted: {
+            activity.start.connect(start)
+            activity.stop.connect(stop)
+        }
+
+        // Add here the QML items you need to access in javascript
+        QtObject {
+            id: items
+            property Item main: activity.main
+            property alias background: background
+            property alias bar: bar
+            property alias bonus: bonus
+            property alias bulbs: bulbs
+            property int sum: 0
+            property int num: 0
+            property int numberOfBulbs: 0
+        }
+
+        onStart: { Activity.start(items) }
+        onStop: { Activity.stop() }
+
+        IntroMessage {
+            id: message
+            onIntroDone: {
+                Activity.initLevel()
+            }
+            intro: [
+                qsTr("Computers use Binary number system, where there are two symbols, 0 and 1."),
+                qsTr("In decimal number system 123 is represented as 1 x 100 + 2 x 10 + 3 x 1"),
+                qsTr("Binary represents numbers in the same pattern, but using powers of 2 instead of powers of 10 that decimal uses"),
+                qsTr("So, 1 in binary is represented by 001, 4 by 100, 7 by 111 and so on.."),
+                qsTr("Our computer has many many switches (called transistors) that can be turned on or off given electricity, and a switch that is on will represent a 1 and a switch that is off will represent a 0."),
+                qsTr("In this activity, you are given a number, you have to find its binary representation by turning on the bulbs. An on bulb representes 1 and an off bulb represents 0")
+            ]
+            z: 20
+            anchors {
+                top: parent.top
+                topMargin: 10
+                right: parent.right
+                rightMargin: 5
+                left: parent.left
+                leftMargin: 5
+            }               
+        }   
+
+        GCText {
+            id: question
+            anchors.horizontalCenter: parent.horizontalCenter
+            anchors.top: parent.top
+            anchors.topMargin: 10
+            fontSize: largeSize
+            width: parent.width * 0.9
+            horizontalAlignment: Text.AlignHCenter
+            wrapMode: Text.WordWrap
+            font.weight: Font.DemiBold
+            style: Text.Outline
+            styleColor: "black"
+            color: "white"
+            text: qsTr("What is the binary representation of %1").arg(items.num)
+            opacity: 1.0
+        }
+
+        Row {
```

Comment:
It is not screen size dependant. On bigger screens, a lot of space will be wasted and on low screens, I think some of the bulbs may be hidden